From: Felix Fietkau <nbd@nbd.name>
Date: Fri, 27 Apr 2018 14:42:14 +0200
Subject: [PATCH] netfilter: nf_flow_table: rework private driver data

Move the timeout out of the union, since it can be shared between the
driver and the stack. Add a private pointer that the driver can use to
point to its own data structures

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---

--- a/include/net/netfilter/nf_flow_table.h
+++ b/include/net/netfilter/nf_flow_table.h
@@ -81,9 +81,10 @@ struct flow_offload_tuple_rhash {
 struct flow_offload {
 	struct flow_offload_tuple_rhash		tuplehash[FLOW_OFFLOAD_DIR_MAX];
 	u32					flags;
+	u32					timeout;
 	union {
 		/* Your private driver data here. */
-		u32		timeout;
+		void *priv;
 	};
 };
 
